addons/mail/models/ir_actions.py

```diff
@@ -51,6 +66,23 @@ def run_action_email(self, action, eval_context=None):
         action.template_id.send_mail(self._context.get('active_id'), force_send=False, raise_exception=False)
         return False
 
+    @api.model
+    def run_action_next_activity(self, action, eval_context=None):
+        if not action.activity_type_id or not self._context.get('active_id'):
+            return False
+
+        record = self.env[action.model_id.model].browse(self._context.get('active_id'))
+        vals = {
+            'date_deadline': action.activity_date_deadline,
+            'summary': action.activity_summary or action.activity_type_id.summary or '',
```

Comment:
I would use an onchange, because if I change the activity type summary it will not give the same result as when the actiob was created. Moreover if I want a void summary it should not take the one from activity type.

Reply:
Actually there is already a fallback in the mixin. More logic is probably not necessary.